modules/core/src/main/scala/org/tessellation/infrastructure/dag/DAGService.scala

```diff
@@ -34,9 +35,10 @@ object DAGService {
           _.map(getBalance(_, address))
         }
 
+      // TODO This needs tests; I can break it with the wrong `filter` and no tests fail
       private def getTotalSupply(snapshot: Signed[GlobalSnapshot]): (BigInt, SnapshotOrdinal) = {
         val empty = BigInt(Balance.empty.coerce.value)
-        val supply = snapshot.value.info.balances.values
+        val supply = snapshot.value.info.balances.filterNot { case (a, _) => lockedAddresses.contains(a) }.values
```

Comment:
fyi - it just filters out the balance for total supply endpoint but it does not change the snapshot which in fact will have an increased total supply. IMO the change should be reflected in the snapshot itself to keep the total supply at the same level, not just on the visual layer